src/aioquic/tls.py

```diff
@@ -551,7 +570,12 @@ def pull_extension() -> None:
                 hello.psk_key_exchange_modes = pull_list(buf, 1, buf.pull_uint8)
             elif extension_type == ExtensionType.SERVER_NAME:
                 with pull_block(buf, 2):
-                    assert buf.pull_uint8() == 0
+                    name_type = buf.pull_uint8()
+                    if name_type != 0:
+                        # We don't know this name_type.
+                        raise AlertIllegalParameter(
+                            f"unknown server name type {name_type}"
+                        )
```

Comment:
I would suggest adding a `pull_server_name` method to facilitate testing. For symmetry you might also consider adding `push_server_name` too.

Reply:
.. and done